akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSinkSpec.scala

```diff
@@ -264,6 +264,22 @@ class InputStreamSinkSpec extends StreamSpec(UnboundedMailboxConfig) {
         inputStream.read()
       }
     }
+
+    "propagate error to InputStream" in {
+      val (probe, inputStream) = TestSource.probe[ByteString]
+        .toMat(StreamConverters.asInputStream(3.seconds))(Keep.both)
+        .run()
+
+      probe.sendNext(ByteString("one"))
+      val error = new RuntimeException("failure")
+      probe.sendError(error)
+      val buffer = Array.ofDim[Byte](5)
+      val thrown = intercept[IOException] {
+        while (true)
```

Comment:
Could that hang if the exception is not thrown? Could we use ScalaTest's `eventually` (or something similar) instead?

Reply:
I've never seen it hang but will add something as a precaution.

Reply:
It shouldn't possibly hang AFACS, it either times out inside of read and throws an IOE because of that, or it sees the failure and throws an IOE because of that

Reply:
read should return -1 if end of stream, so in the case of an a bug in our code this would be an infinite spin loop. Perhaps:
```
inputStream.read(buffer) should !==(-1)
```